### DISCUSSION TITLE
[Contract Verification] Vyper Verification

### DISCUSSION THREAD

**Discussion by PatrickAlphaC** [zkSync-Community-Hub/zksync-developers#681] [Contract Verification] [answered]
### Environment

Testnet

### Block Explorer

https://explorer.zksync.io/

### Issue Type

- [ ] Plugin
- [X] Block Explorer

### Contract Address

doesnt matter

### Compiler Type

Single file

### zkSolc Version

n/a

### Solc Version

n/a

### Contract Name

n/a

### Contract Code

n/a

### Constructor Arguments

n/a

### Additional Details

https://github.com/matter-labs/block-explorer/issues/285

It looks like vyper work on the API for the block explorer. Does anyone have an example of this per chance?

**Comment by alhenaai** [comment]
To verify Vyper contracts on the ZKsync Era network using the Hardhat plugin, you need to ensure that the Vyper smart contracts list in your project's "contracts" folder is exactly the same as it was during the deployment of the specific Vyper contract. This is crucial because any discrepancy will result in a "bytecode doesn't match any of your local contracts" error. It's strongly recommended to verify your Vyper smart contracts immediately after their deployment to minimize such issues.

For more detailed guidance on using the Hardhat plugin for Vyper verification, you can refer to the [hardhat-zksync-verify-vyper documentation](https://docs.zksync.io/build/tooling/hardhat/plugins/hardhat-zksync-verify-vyper).

If you are looking for examples of Vyper work on the API for the block explorer, unfortunately, I couldn't find specific examples in the provided documentation. You might want to check the official repositories or community forums for more detailed examples or guidance.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Reply by bxpana** [reply, marked as answer]
Here's an example of how the request looks:
```
'{"contractAddress":"0x570ecE11BAA365E2109498Ab3D0dd44D199F106C","sourceCode":{"contracts/CreateForwarder.vy":"# @version ^0.3.3\\n# vim: ft=python\\n\\ninterface Greeter:\\n    def setup(name: String[100]): nonpayable\\n\\nforwarder: public(address)\\n\\n@external\\ndef deploy(_masterCopy: address, _greeting: String[100]):\\n    self.forwarder = create_forwarder_to(_masterCopy)\\n    # Greeter.__init__ was not called, else this would fail\\n    Greeter(self.forwarder).setup(_greeting)\\n","contracts/Greeter.vy":"# @version ^0.3.3\\n# vim: ft=python\\n\\nowner: public(address)\\ngreeting: public(String[100])\\n\\n# __init__ is not called when deployed from create_forwarder_to\\n@external\\ndef __init__():\\n  self.owner = msg.sender\\n  self.greeting = \\"Hola mundo\\"\\n\\n# call once after create_forwarder_to\\n@external\\ndef setup(_greeting: String[100]):\\n  assert self.owner == ZERO_ADDRESS, \\"owner != zero address\\"\\n  self.owner = msg.sender\\n  self.greeting = _greeting\\n\\n@external\\n@view\\ndef greet() -> String[100]:\\n    return self.greeting\\n"},"codeFormat":"vyper-multi-file","contractName":"Greeter","compilerVyperVersion":"0.3.9","compilerZkvyperVersion":"v1.5.3","constructorArguments":"0x","optimizationUsed":true}'
```
